src/Command/SettingsDebugCommand.php

```diff
@@ -0,0 +1,190 @@
+        ])->setDescription('Displays current settings for an application')->setHelp(<<<'EOF'
+The <info>%command.name%</info> command displays all available settings:
+
+  <info>php %command.full_name%</info>
+
+Use the <info>--domains</info> option to display all configured domains:
+
+  <info>php %command.full_name% --domains</info>
+
+Display settings for a specific domain by specifying its name with the <info>--domain</info> option:
+
+  <info>php %command.full_name% --domain=default</info>
+
+Find all settings with a specific tag by specifying the tag name with the <info>--tag</info> option:
+
+  <info>php %command.full_name% --tag=foo</info>
+
+EOF
+        );
+    }
+
+    /**
+     * Executes the current command.
+     * This method is not abstract because you can use this class
+     * as a concrete class. In this case, instead of defining the
+     * execute() method, you set the code to execute by passing
+     * a Closure to the setCode() method.
+     *
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return void null or 0 if everything went fine, or an error code
+     * @see setCode()
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        // // Displays all configured domains
+        if ($input->getOption('domains')) {
+            $tableHeaders = ['Name', 'Priority', 'Enabled', 'Read only'];
+            $tableRows    = [];
+            foreach ($this->settingsManager->getDomains() as $key => $domain) {
+                $tableRows[] = [
+                    $domain->getName(),
+                    $domain->getPriority(),
+                    $domain->isEnabled(),
+                    $domain->isReadOnly()
+                ];
+            }
+            $io->table($tableHeaders, $tableRows);
+        }
+
+        // Filter by domain(s)
+        $domains = array_keys($this->settingsManager->getDomains());
+        if ($input->getOption('domain')) {
+            $domains = [$input->getOption('domain')];
+        }
+
+        // Displays table of settings
+        if (!$input->getOption('domains') && !$input->getArgument('name')) {
+            $tableHeaders = ['Name', 'Description', 'Domain', 'Type', 'Data', 'Provider name', 'Tags'];
+            $tableRows    = [];
+
+            // Display settings filtered by tags
+            if ($tag = $input->getOption('tag')) {
+                foreach (array_reverse($this->settingsManager->getEnabledSettingsByTag($domains,
+                    $tag)) as $settingModel) {
+                    $tableRows[] = $this->_renderSettingsRow($settingModel);
+                }
+            } // Display all settings
+            else {
+                /** @var SettingsProviderInterface $provider */
+                foreach (array_reverse($this->settingsManager->getProviders()) as $pName => $provider) {
```

Comment:
Null provider names are possible due to this. These values is being set by SettingsManager. Options here:
 - set provider name manually to setting model just below few lines
 - use settings manager `getSettingsByDomain` but then settings will be merged.